sotodlib/core/metadata/loader.py

```diff
@@ -71,229 +85,526 @@ def load_raw(self, spec_list, request, detdb=None):
 
-        Instead, this function returns information about whether
-        metadata caused any trouble.
+                det_info, aug_request = check_tags(det_info, aug_request)
 
-        """
-        errors = []
-        ok = True
-        for spec in spec_list:
+            if check:
+                items.append((spec, error))
+                continue
+
+            if item is None:
+                # Exit for the det_info case.
+                continue
+
+            # Make everything an axisman.
+            if isinstance(item, ResultSet):
+                item = broadcast_resultset(item, det_info=det_info)
+
+            elif not isinstance(item, core.AxisManager):
+                logger.error(
+                    f'The decoded item {item} is not an AxisManager or '
+                    f'other well-understood type.  Request was: {request}.')
+
+            # Unpack it.
             try:
-                item = self.load_raw([spec], request)
-                errors.append((spec, None))
+                unpackers = Unpacker.decode(spec['name'], item)
+                for unpacker in unpackers:
+                    dest = unpacker.unpack(item, dest=dest)
             except Exception as e:
-                errors.append((spec, e))
-                ok = False
-        return ok, errors
+                reraise(spec, e)
+
+            logger.debug(f'load(): dest now has shape {dest.shape}')
+
+        check_tags(det_info, aug_request, final=True)
+
+        if check:
+            return items
+
+        dest.wrap('det_info', convert_det_info(det_info))
+
+        return dest
+
+
+def _filter_items(prefix, d, remove=True):
+    # Restrict d to only items that start with prefix; if d is a dict,
+    # return a dict with only the keys that satisfy that condition.
+    if isinstance(d, dict):
+        return {k: d[prefix*remove + k]
+                for k in _filter_items(prefix, list(d.keys()), remove=remove)}
+    return [k[len(prefix)*remove:] for k in d if k.startswith(prefix)]
+
+
+def merge_det_info(det_info, new_info,
+                   index_columns=['readout_id', 'det_id']):
+    """Args:
+
+      det_info (ResultSet or None): The det_info table to start from,
+        with columns *without* the 'dets:' prefix.
+      new_info (ResultSet): New data to merge/check against
+        det_info; only columns with dets: prefix are processed.
+
+    Returns:
+      A (possibly) new det_info table, containing updates and
+      restrictions from new_info.  This will contain at least the
+      columns that det_info had, and at most as many rows.
+
+    Notes:
+      The input arguments det_info and new_info may be modified by
+      this function.  Passing in None for det_info is convenient to
+      initialize a det_info from a new_info where the columns are
+      named dets:... .
+
+    """
+    new_keys = _filter_items('dets:', new_info.keys)
+    if (len(new_keys) != len(new_info.keys)):
+        reraise(spec, RuntimeError(
```

Comment:
because I'm good at hitting every one of these errors. I'm getting a "NameError: name 'reraise' is not defined" error here.